### ISSUE TITLE
Reproduce/replace name normalization model with an LLM-based one (alternative to #6)

### ISSUE THREAD

**Issue by jpivarski** [uchicago-dsi/good-food-purchasing-nova-classification#5]
CGFP has been using https://github.com/uchicago-dsi/good-food-purchasing to convert raw food product names with a normalized name that's better for further processing. However, it was delivered to CGFP as a Google Colab notebook in Python 3.10, and Google Colab no longer runs on Python 3.10 ([Medium](https://medium.com/google-colab/colab-updated-to-python-3-11-00197d6172b1); release notes say 3.10 → 3.11 on 2025-04-29, 3.11 → 3.12 on 2025-08-27). All package versions would need to be updated, and that means retraining. In addition, CGFP has other reasons to update the training data.

To deploy in a way that is less sensitive to such changes (only changes in CGFP's own training data), we could reproduce the name normalization model with a ChatGPT fine-tuning, similar to but distinct from the NOVA classification.

**Steps in this process:**
1. reproduce the original model in a Docker container with pinned package versions
2. develop a fine-tuned model that performs the same task with the same input data, which can be found here: [Google Drive](https://drive.google.com/drive/u/1/folders/1KbhIw2XmHR806nMClO4NhTkIVOBSr93F)
3. update with CGFP's data

**Acceptance criteria:**
* demonstration that the new model has similar or higher accuracy than the old
* make it public
* provide a script that runs it
